lib/core/src/Cardano/Wallet.hs

```diff
@@ -1684,6 +1619,80 @@ balanceTransaction
+        --  2afeed9b
+        --  []
+        --  inputs 2nd 01f4b788
+        --  outputs address: 82d81858...6f57b300
+        --          coin: 0.000000
+        --          tokens: []
+        --  []
+        --  metadata:
+        --  scriptValidity: valid
+
+        --  Lovelace 1000000 /= Lovelace 0
+        --
+        --  This is probably due to selectAssets replacing 0 ada outputs with
+        --  minUTxOValue in the selection, which doesn't end up in the CBOR tx.
+        let zeroAdaOutputs =
+                filter (\o -> view (#tokens . #coin) o == Coin 0 ) outputs
+        unless (null zeroAdaOutputs) $
+            throwE $ ErrBalanceTxNotYetSupported ZeroAdaOutput
+
+    guardDeposits = do
+        -- There is currently no way of telling 'selectAssets' about all the
+        -- deposits and refunds in the transaction. Not via 'TransactionCtx'.
+        --
+        -- A promising fix would be to replace the details of 'TransactionCtx' with
+        -- a (balance, fee) based on calling the node/ledger
+        -- @evaluateTransactionBalance@ on the partial transacion.
+        let isReg (Cardano.StakePoolRegistrationCertificate _) = True
+            isReg (Cardano.StakeAddressRegistrationCertificate _) = True
+            isReg (Cardano.StakeAddressDeregistrationCertificate _) = True
+            isReg _ = False
+        case Cardano.txCertificates bod of
+            Cardano.TxCertificatesNone -> return ()
+            Cardano.TxCertificates _ certs _
+                | any isReg certs -> throwE $ ErrBalanceTxNotYetSupported Deposits
+                | otherwise -> return ()
+
+    guardConflictingWithdrawalNetworks = do
+        -- Use of withdrawals with different networks breaks balancing.
+        --
+        -- For instance the partial tx might contain two withdrawals with the same
+        -- key but different networks:
+        -- [ (Mainnet, pkh1, coin1)
+        -- , (Testnet, pkh1, coin2)
+        -- ]
+        --
+        -- Even though this is absurd, the node/ledger @evaluateTransactionBalance@
+        -- will count @coin1+coin2@ towards the total balance. Because the wallet
+        -- does not consider the network tag, it will drop one of the two, leading
+        -- to a discrepancy.
+        let networkOfWdrl ((Cardano.StakeAddress nw _), _, _) = nw
+        let conflictingWdrlNetworks = case Cardano.txWithdrawals bod of
+                Cardano.TxWithdrawalsNone -> False
+                Cardano.TxWithdrawals _ wdrls -> Set.size
+                    (Set.fromList $ map networkOfWdrl wdrls) > 1
+        when conflictingWdrlNetworks $
+            throwE $ ErrBalanceTxNotYetSupported ConflictingNetworks
+
+    guardExistingCollateral = do
+        -- Coin selection does not support pre-defining collateral. In Sep 2021
+        -- consensus was that we /could/ allow for it with just a day's work or so,
+        -- but that the need for it was unclear enough that it was not in any way
+        -- a priority.
```

Comment:
I might be wrong, but suspect it would not just be "a day's work" 😄 .

Reply:
😄 I thought that was what you said